fix: update to react-router v6

coderbyheart opened the pull request:
Fixes #426

coderbyheart committed:
fix: update to react-router v6

ghost commented:
### CodeSee Review Map:
[![Review these changes using an interactive CodeSee Map](https://s3.us-east-2.amazonaws.com/maps.codesee.io/images/github/distributeaid/shipment-tracker/519/6a6812e1/3dccbef787e32382ef46ae24f3e77bc558c7164e.svg)](https://app.codesee.io/r/reviews?pr=519&src=https%3A%2F%2Fgithub.com%2Fdistributeaid%2Fshipment-tracker)

Review [in an interactive map](https://app.codesee.io/r/reviews?pr=519&src=https%3A%2F%2Fgithub.com%2Fdistributeaid%2Fshipment-tracker)

View [more CodeSee Maps](https://app.codesee.io/r/maps?src=https%3A%2F%2Fgithub.com%2Fdistributeaid%2Fshipment-tracker)

#### Legend
![CodeSee Map Legend](https://app.codesee.io/Legend_v2.png)

coderbyheart committed:
fix: simplify routing

This removes the logic for handling redirects to
inaccessible pages after login using the `redirectAfterAuth`
route state, which was not implemented anyway.

This simplifies the definition of the routes.

coderbyheart committed:
fix: remove types
